presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/TypeUtils.java

```diff
@@ -238,6 +262,19 @@ private static Object prestoNativeToJdbcObject(ConnectorSession session, Type pr
             return new Date(UTC.getMillisKeepLocal(DateTimeZone.getDefault(), millis));
         }
 
+        if (TIMESTAMP.equals(prestoType)) {
+            if (session.isLegacyTimestamp()) {
+                ZoneId sessionZone = ZoneId.of(session.getTimeZoneKey().getId());
+                return fromPrestoLegacyTimestamp((long) prestoNative, sessionZone);
+            }
+            return fromPrestoTimestamp((long) prestoNative);
```

Comment:
This returns `LocalDateTime` while `jdbcObjectToPrestoNative` expects `j.s.Timestamp`.

- Does the driver handle both types when writing?
- Can we somehow force the driver to get arrays as  `LocalDateTime` instances? Going thru `j.s.Timestamp` is inherently incorrect in certain edge cases.

Reply:
Looks like it's possible:
https://jdbc.postgresql.org/documentation/head/8-date-time.html

Reply:
I didn't find a way to force the driver to get arrays as `LocalDateTime`, so I changed `prestoNativeToJdbcObject` to return `j.s.Timestamp`